packages/ember-glimmer/tests/integration/components/append-test.js

```diff
@@ -28,20 +28,226 @@ class AbstractAppendTest extends RenderingTest {
+    }
+
+    this.registerComponent('x-parent', {
+      ComponentClass: Component.extend({
+        layoutName: 'components/x-parent'
+      }),
+
+      template: '[parent: {{foo}}]{{#x-child bar=foo}}[yielded: {{foo}}]{{/x-child}}'
+    });
+
+    this.registerComponent('x-child', {
+      ComponentClass: Component.extend({
+        tagName: ''
+      }),
+
+      template: '[child: {{bar}}]{{yield}}'
+    });
+
+    let XParent = this.owner._lookupFactory('component:x-parent');
+
+    this.component = XParent.create({ foo: 'zomg' });
+
+    assert.deepEqual(hooks, [
+      ['x-parent', 'init'],
+      ['x-parent', 'didInitAttrs'],
+      ['x-parent', 'didReceiveAttrs'],
+      ['x-parent', 'on(init)']
+    ], 'creation of x-parent');
+
+    hooks.length = 0;
+
+    this.element = this.append(this.component);
+
+    assert.deepEqual(hooks, [
+      ['x-parent', 'willInsertElement'],
+      ['x-parent', 'willRender'],
+
+      ['x-child', 'init'],
+      ['x-child', 'didInitAttrs'],
+      ['x-child', 'didReceiveAttrs'],
+      ['x-child', 'on(init)'],
+      ['x-child', 'willInsertElement'],
+      ['x-child', 'willRender'],
+
+      ['x-child', 'didInsertElement'],
+      ['x-child', 'didRender'],
+
+      ['x-parent', 'didInsertElement'],
+      ['x-parent', 'didRender']
+    ], 'appending of x-parent');
+
+    hooks.length = 0;
+
+    let componentElement = this.component.element;
+
+    this.runTask(() => this.rerender());
+
+    assert.deepEqual(hooks, [
+      ['x-parent', 'willUpdate'],
+      ['x-parent', 'willRender'],
+
+      ['x-child', 'willUpdate'],
+      ['x-child', 'willRender'],
+
+      ['x-child', 'didUpdate'],
+      ['x-child', 'didRender'],
+
+      ['x-parent', 'didUpdate'],
+      ['x-parent', 'didRender']
+    ], 'rerender');
+
+    hooks.length = 0;
+
+    this.runTask(() => set(this.component, 'foo', 'wow'));
+
+    assert.deepEqual(hooks, [
+      ['x-parent', 'willUpdate'],
+      ['x-parent', 'willRender'],
+
+      ['x-child', 'didUpdateAttrs'],
+      ['x-child', 'didReceiveAttrs'],
+
+      ['x-child', 'willUpdate'],
+      ['x-child', 'willRender'],
+
+      ['x-child', 'didUpdate'],
+      ['x-child', 'didRender'],
+
+      ['x-parent', 'didUpdate'],
+      ['x-parent', 'didRender']
+    ], 'set foo = wow');
+
+    hooks.length = 0;
+
+    this.runTask(() => set(this.component, 'foo', 'zomg'));
+
+    assert.deepEqual(hooks, [
+      ['x-parent', 'willUpdate'],
+      ['x-parent', 'willRender'],
+
+      ['x-child', 'didUpdateAttrs'],
+      ['x-child', 'didReceiveAttrs'],
+
+      ['x-child', 'willUpdate'],
+      ['x-child', 'willRender'],
+
+      ['x-child', 'didUpdate'],
+      ['x-child', 'didRender'],
+
+      ['x-parent', 'didUpdate'],
+      ['x-parent', 'didRender']
+    ], 'set foo = zomg');
+
+    hooks.length = 0;
+
+    this.runTask(() => this.component.destroy());
+
+    assert.deepEqual(hooks, [
+      [ 'x-parent', 'didDestroyElement' ],
```

Comment:
Manually destroying components likely just doesn't have the right lifecycle events.  I suspect it is not your test.